lib/ansible/galaxy/api.py

```diff
@@ -893,7 +897,9 @@ def get_collection_versions(self, namespace, name):
 
         versions = []
         while True:
-            versions += [v['version'] for v in data[results_key]]
+            for v in data[results_key]:
+                versions.append(v["version"])
+                self.requires_ansible.setdefault(f"{namespace}.{name}", {})[v["version"]] = v.get("requires_ansible")
```

Comment:
With `defaultdict` suggested above, this line can look simpler:
```suggestion
                self.requires_ansible[f"{namespace}.{name}"][v["version"]] = v.get("requires_ansible")
```